README.md

```diff
@@ -52,139 +79,208 @@ let addrData = await addRawData.json()
+  params: [senderHexAddress, eip712Payload],
 })
 
-// The chain and sender objects are the same as the previous example
-let extension = signatureToWeb3Extension(chain, sender, signature)
+// Create a signed Tx payload that can be broadcast to a node.
+const signatureBytes = Buffer.from(signature.replace('0x', ''), 'hex')
 
-// Create the txRaw
-let rawTx = createTxRawEIP712(
-  msg.legacyAmino.body,
-  msg.legacyAmino.authInfo,
-  extension,
-)
-
-// Broadcast it
-const postOptions = {
-  method: 'POST',
-  headers: { 'Content-Type': 'application/json' },
-  body: generatePostBodyBroadcast(rawTx),
-}
+const { signDirect } = tx
+const bodyBytes = signDirect.body.toBinary()
+const authInfoBytes = signDirect.authInfo.toBinary()
 
-let broadcastPost = await fetch(
-  `http://localhost:1317${generateEndpointBroadcast()}`,
-  postOptions,
+const signedTx = createTxRaw(
+  bodyBytes,
+  authInfoBytes,
+  [signatureBytes],
 )
-let response = await broadcastPost.json()
 ```
 
-### Signing with Keplr
+### Sign the Transaction with Keplr (SignDirect)
+
+EvmosJS supports Cosmos SDK `SignDirect` payloads that can be signed using Keplr.
 
 ```ts
-// Follow the previous step to generate the msg object
 import { createTxRaw } from '@evmos/proto'
-import {
-  generateEndpointBroadcast,
-  generatePostBodyBroadcast,
-} from '@evmos/provider'
 
-let sign = await window?.keplr?.signDirect(
+// First, populate a TxContext object and create a signable Tx payload.
+const context = ...
+const tx = ...
+
+const { chain, sender } = context
+const { signDirect } = tx
+
+const signResponse = await window?.keplr?.signDirect(
   chain.cosmosChainId,
   sender.accountAddress,
   {
-    bodyBytes: msg.signDirect.body.serializeBinary(),
-    authInfoBytes: msg.signDirect.authInfo.serializeBinary(),
+    bodyBytes: signDirect.body.toBinary(),
+    authInfoBytes: signDirect.authInfo.toBinary(),
     chainId: chain.cosmosChainId,
     accountNumber: new Long(sender.accountNumber),
   },
-  // @ts-expect-error the types are not updated on Keplr side
-  { isEthereum: true },
 )
 
-if (sign !== undefined) {
-  let rawTx = createTxRaw(sign.signed.bodyBytes, sign.signed.authInfoBytes, [
-    new Uint8Array(Buffer.from(sign.signature.signature, 'base64')),
-  ])
+if (!signResponse) {
+  // Handle signature failure here.
+}
+
+const signatures = [
+  new Uint8Array(Buffer.from(signResponse.signature.signature, 'base64')),
+]
 
-  // Broadcast it
-  const postOptions = {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    body: generatePostBodyBroadcast(rawTx),
-  }
+const { signed } = signResponse
+
+const signedTx = createTxRaw(
+  signed.bodyBytes,
+  signed.authInfoBytes,
+  signatures,
+])
+```
+
+### Sign the Transaction with Keplr (EIP-712)
+
+EvmosJS also supports signing EIP-712 payloads using Keplr. This is necessary for Ledger users on Keplr, since the Ledger cannot sign SignDirect payloads.
```

Comment:
```suggestion
EvmosJS also supports signing [EIP-712](https://eips.ethereum.org/EIPS/eip-712) payloads using Keplr. This is necessary for Ledger users on Keplr, since the Ledger device cannot sign `SignDirect` payloads.
```